tests/Tests.FeatureManagement/FeatureManagement.cs

```diff
@@ -483,5 +491,59 @@ public async Task SwallowsExceptionForMissingFeatureFilter()
 
             Assert.False(isEnabled);
         }
+
+        [Fact]
+        public async Task ThrowsOnMissingFeature()
+        {
+            IConfiguration config = new ConfigurationBuilder().AddJsonFile("appsettings.json").Build();
+
+            var services = new ServiceCollection();
+
+            services
+                .Configure<FeatureManagementOptions>(options =>
+                {
+                    options.IgnoreMissingFeatures = false;
+                });
+
+            services
+                .AddSingleton(config)
+                .AddFeatureManagement()
+                .AddFeatureFilter<TestFilter>();
+
+            ServiceProvider serviceProvider = services.BuildServiceProvider();
+
+            IFeatureManager featureManager = serviceProvider.GetRequiredService<IFeatureManager>();
+
+            FeatureManagementException e = await Assert.ThrowsAsync<FeatureManagementException>(async () => await featureManager.IsEnabledAsync(MissingFeature));
+
+            Assert.Equal(FeatureManagementError.MissingFeature, e.Error);
+        }
+
+        [Fact]
+        public async Task SwallowsOnMissingFeature()
+        {
+            IConfiguration config = new ConfigurationBuilder().AddJsonFile("appsettings.json").Build();
+
+            var services = new ServiceCollection();
+
+            services
+                .Configure<FeatureManagementOptions>(options =>
+                {
+                    options.IgnoreMissingFeatures = true;
```

Comment:
I would remove this so we can test the default behavior and ensure no regressions.